src/views/Proposal.vue

```diff
@@ -75,7 +75,7 @@ async function loadProposal() {
   proposal.value = proposalObj.proposal;
 
   // Redirect to proposal spaceId if it doesn't match route key
-  if (route.key !== proposal.value.space.id)
+  if (route.params.key && route.params.key !== proposal.value.space.id)
```

Comment:
If you click to open a proposal then quickly change space it will redirect you to the post, this is not the expected behavior, we should check maybe for route name too